src/mmrelay/migrate.py

```
@@ -0,0 +1,2044 @@
+    # Copy-then-delete pattern: Always copy first, verify, then delete sources only if verification succeeds.
+    # This prevents data loss if integrity check fails after files are moved.
+    if not force:
+        for db_path in selected_group:
+            dest = new_db_dir / db_path.name
+            if dest.exists():
+                logger.info("Backing up existing database: %s", dest)
+                backup_path = _backup_file(dest)
+                try:
+                    shutil.copy2(str(dest), str(backup_path))
+                except (OSError, IOError) as e:
+                    logger.error("Failed to backup database %s: %s", dest, e)
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)